### ISSUE TITLE
newrelic_entity data source cannot lookup AWS Billing and AWS Lambda Function and more...

### ISSUE THREAD

**Issue by silanpa** [newrelic/terraform-provider-newrelic#2229]
### Terraform Version
```bash
Terraform v1.3.4
on darwin_arm64
+ provider registry.terraform.io/newrelic/newrelic v3.7.1

Your version of Terraform is out of date! The latest version
is 1.3.7. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)
- `newrelic_entity` data resource

### Terraform Configuration
```hcl
provider "newrelic" {
  account_id = var.newrelic_account_id
  api_key    = var.newrelic_apikey
  region     = var.newrelic_region
}

terraform {
  required_providers {
    newrelic = {
      source = "newrelic/newrelic"
    }
  }
}
```
```hcl
data "newrelic_entity" "infra_budget_data" {
  name = "Infra server"
  domain = "INFRA"
  type = "AWSBILLINGBUDGET"
}

data "newrelic_entity" "infra_container_data" {
  name = "Infra Container"
  domain = "INFRA"
  type = "AWSECSCONTAINERINSTANCE"
}

data "newrelic_entity" "lambda_data" {
  name = "Lambda Test"
  domain = "INFRA"
  type = "AWSLAMBDAFUNCTION"
}
```

### Error produce(s)

```hcl
 Error: expected type to be one of [APPLICATION DASHBOARD HOST MONITOR WORKLOAD], got AWSBILLINGBUDGET
│ 
│   with data.newrelic_entity.infra_budget_data,
│   on data_entity.tf line 4, in data "newrelic_entity" "infra_budget_data":
│    4:   type = "AWSBILLINGBUDGET"
│ 
╵
╷
│ Error: expected type to be one of [APPLICATION DASHBOARD HOST MONITOR WORKLOAD], got AWSECSCONTAINERINSTANCE
│ 
│   with data.newrelic_entity.infra_container_data,
│   on data_entity.tf line 10, in data "newrelic_entity" "infra_container_data":
│   10:   type = "AWSECSCONTAINERINSTANCE"
│ 
╵
╷
│ Error: expected type to be one of [APPLICATION DASHBOARD HOST MONITOR WORKLOAD], got AWSLAMBDAFUNCTION
│ 
│   with data.newrelic_entity.lambda_data,
│   on data_entity.tf line 16, in data "newrelic_entity" "lambda_data":
│   16:   type = "AWSLAMBDAFUNCTION"
```

### GraphQl queries

** NOTE:** Some values ​​were redacted to protect private and sensitive data!

```graphql
{
  actor {
    entitySearch(queryBuilder: {domain: INFRA, name: "Infra Container"}) {
      query
      results {
        entities {
          entityType
          name
          type
          guid
        }
      }
    }
  }
}
```

```json
{
  "data": {
    "actor": {
      "entitySearch": {
        "query": "domain IN ('INFRA') AND name LIKE 'Infra Container'",
        "results": {
          "entities": [
            {
              "entityType": "INFRASTRUCTURE_AWS_LAMBDA_FUNCTION_ENTITY",
              "guid": "REDACTED",
              "name": "Infra Container",
              "type": "AWSLAMBDAFUNCTION"
            }
          ]
        }
      }
    }
  }
}
```

### Feature Description
Add all type posible related to GraphQL suported for use in provider and could find entitities.

**Comment by kidk** [comment]
Can you try removing the type @silanpa? We have some documentation improvements ready to make this clearer.

**Comment by kidk** [comment]
@silanpa I've reached out internally as the `instructureIntegrationType` field which we would use for these kind of entities is deprecated. The current workaround would be to move the `type` but I understand it's not ideal. 

I'll update here once I get some more details on the path forward.

**Comment by kidk** [comment]
We are going to rework this resource to use the `query` instead of `querybuilder` endpoints of our API. This would allow us to do any sort of filtering and would make the resource less dependant on filtering with tags. 

We need to be careful not to break backwards compatibility so it's going to take some time. We expect to start on this work in the coming months. 

In the meantime it's best to remove the type. Please let us know if you encounter any other limitations so we can keep those in mind when reworking the resource.

**Comment by sanderblue** [comment]
Hi @silanpa, we recently released an update to how we query for an entity in the `newrelic_entity` data source.

Can you try again with the latest version [**v3.16.1**](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v3.16.1)?

As mentioned before, please try the data source _without_ the `type` attribute first if possible, but if it's necessary, feel to add it. Either way, hopefully you get the desired results. Please let us know if this is working for you after upgrading. Thanks! 🙂

**Comment by sanderblue** [comment]
Closing since we haven't heard back in a while. Based on #2135, this issue is also likely fixed in v3.16.1+
